reindex-solr is failing

<!-- What problem are we solving? What does the experience look like today? What are the symptoms? -->

reindex-solr is failing and this causes no books to show up in dev environment

### Steps to Reproduce
<!-- What steps caused you to find the bug? -->
1. Open master branch and run `docker-compose run --rm home make reindex-solr`

<!-- What actually happened after these steps? What did you expect to happen? -->
* Actual: failure
* Expected: success

Error message:
```
gitpod /workspace/openlibrary $ docker-compose run --rm home make reindex-solr
Creating openlibrary_home_run ... done
psql --host db openlibrary -t -c 'select key from thing' | sed 's/ *//' | grep '^/books/' | PYTHONPATH=/openlibrary xargs python openlibrary/solr/update_work.py --ol-url http://web:8080/ --ol-config conf/openlibrary.yml --data-provider=legacy
Traceback (most recent call last):
  File "/openlibrary/openlibrary/solr/update_work.py", line 1622, in <module>
    FnToCLI(main).run()
  File "/openlibrary/scripts/solr_builder/solr_builder/fn_to_cli.py", line 51, in __init__
    arg_opts = self.type_to_argparse(annotations[arg])
  File "/openlibrary/scripts/solr_builder/solr_builder/fn_to_cli.py", line 109, in type_to_argparse
    raise ValueError(f'Unsupported type: {typ}')
ValueError: Unsupported type: typing.List[str]
Makefile:62: recipe for target 'reindex-solr' failed
make: *** [reindex-solr] Error 123
ERROR: 2
```

### Details

- **Logged in (Y/N)?** N
- **Browser type/version?** FF
- **Operating system?** Mac
- **Environment (prod/dev/local)?** local
<!-- If not sure, put prod -->

### Proposal & Constraints
<!-- What is the proposed solution / implementation? Is there a precedent of this approach succeeding elsewhere? -->

### Related files
<!-- Files related to this issue; this is super useful for new contributors who might want to help! If you're not sure, leave this blank; a maintainer will add them. -->

Seems to be related to this change: https://github.com/internetarchive/openlibrary/pull/5841/files#diff-4e97b24de087da37962d5635500c82dcb24afde814ea3ef13527d0995f0a5fe5R103


### Stakeholders
<!-- @ tag stakeholders of this bug -->
@cdrini

--- a/openlibrary/solr/update_work.py
+++ b/openlibrary/solr/update_work.py
@@ -1575,7 +1575,7 @@ def get_ia_db(settings):
 
 
 def main(
-        keys: List[str],
+        keys: list[str],
         ol_url="http://openlibrary.org",
         ol_config="openlibrary.yml",
         output_file: str = None,

